src/lib/source-handlers/azure/list-repos.ts

```diff
@@ -31,6 +31,21 @@ export async function fetchAllRepos(
   token: string,
 ): Promise<AzureRepoData[]> {
   debug('Fetching repos for ' + project);
+  let repoList: AzureRepoData[] = [];
+  try {
+    repoList = await getRepos(url, orgName, project, token);
+  } catch (err) {
+    throw new Error(JSON.stringify(err));
```

Comment:
there is no need to catch & throw here again, you threw in the function called then let this simply bubble all the way up. You can remove this try/catch

Reply:
The lower throw is only for when the call was sent and we did not get a 200 response. This throw is for when the call is not even sent.